bin/update-ips.bash

```diff
@@ -744,34 +731,112 @@ validate_config() {
   fi
 }
 
-validate_config
-#TODO: To be changed when decision made on networkpolicies for azure storage
-if [ "$storage_service" == "azure" ]; then
-  :
-else
-  allow_object_storage
+check_cluster="${1}"
+[[ "${1}" =~ ^(wc|sc|both)$ ]] || log_fatal "first argument must be either wc, sc or both"
+
+dry_run=true
+if [[ "${2}" == "apply" ]]; then
+  dry_run=false
 fi
-allow_ingress
 
-if [[ "${check_cluster}" =~ ^(sc|both)$ ]]; then
-  allow_subnet "sc" '.networkPolicies.global.scApiserver.ips' "node-role.kubernetes.io/control-plane="
-  allow_subnet "sc" '.networkPolicies.global.scNodes.ips' ""
+shift 2
 
-  if swift_enabled; then
-    sync_swift '.objectStorage.swift' '.networkPolicies.global.objectStorageSwift'
-  fi
-fi
+has_diff=0
 
-if [[ "${check_cluster}" =~ ^(wc|both)$ ]]; then
-  allow_subnet "wc" '.networkPolicies.global.wcApiserver.ips' "node-role.kubernetes.io/control-plane="
-  allow_subnet "wc" '.networkPolicies.global.wcNodes.ips' ""
-fi
+# TODO: To be changed when decision made on networkpolicies for azure storage
+storage_service=$(yq '.objectStorage.type' "${CK8S_CONFIG_PATH}/defaults/common-config.yaml")
 
-if rclone_enabled; then
-  sync_rclone '.objectStorage.sync.s3.regionEndpoint' '.networkPolicies.rclone.sync.objectStorage'
-  sync_swift '.objectStorage.sync.swift' '.networkPolicies.rclone.sync.objectStorageSwift'
-  sync_rclone '.objectStorage.sync.secondaryUrl' '.networkPolicies.rclone.sync.secondaryUrl'
-fi
+CK8S_IPV6_ENABLED="false"
+
+declare -A operations=(
+  [objectstorage]=-
+  [ingress]=-
+  [apiserver]=-
+  [nodes]=-
+  [swift]=-
+  [rclone]=-
+)
+
+while [ -n "${1+x}" ]; do
+  case "${1}" in
+  "--enable")
+    if [ "${#}" -lt 2 ]; then
+      log_fatal "The --enable flag requires an operation, example: --enable ingress"
+    fi
+    operations=(["${2}"]=-)
+    shift
+    ;;
+  "--disable")
+    if [ "${#}" -lt 2 ]; then
+      log_fatal "The --disable flag requires an operation, example: --disable ingress"
+    fi
+    unset 'operations[${2}]'
+    shift
+    ;;
+  "--ipv6")
+    CK8S_IPV6_ENABLED="true"
+    ;;
+  *)
+    log_fatal "Unsupported update-ips flag: ${1}"
```

Comment:
Could the available flags be listed, so you can find out without reading the source. Would be nice if it fit into the the `usage` function in `bin/ck8s`, tho it looks a bit crowded 🤔

Reply:
Added `ck8s update-ips both operations` to list the available operations: https://github.com/elastisys/compliantkubernetes-apps/pull/2719/commits/b191e7fb1a117d4d55f63ba446b2d2ec0c8a6705